browsers/duckduckgo.safariextension/js/background.js

```diff
@@ -106,4 +107,59 @@ var onBeforeRequest = function (requestData) {
     }   
 }
 
+
+/**
+ * Before navigating to a new page,
+ * check whether we should upgrade to https
+ */
+var onBeforeNavigation = function (e) {
+    //console.log("beforeNavigation", e.url, e.target.url, e.eventPhase, e)
+
+    if (!e.url) return
+
+    const tabId = tabManager.getTabId(e)
+    let thisTab = tabManager.get({tabId: tabId})
+
+    if (!thisTab) {
+        thisTab = tabManager.create(e)
+        console.log('onBeforeNavigation CREATED TAB:', thisTab)
+    }
+
+    // same logic from /shared/js/background.js
+
+    // site is required to be there:
+    if (!thisTab || !thisTab.site) {
+        console.log('HTTPS: no tab or tab site found for: ', tabId, thisTab)
+        return
+    }
+    
+    // skip upgrading broken sites:
+    if (thisTab.site.isBroken) {
+        console.log('HTTPS: temporarily skip upgrades for: ' + e.url)
+        return
+    }
+
+    // avoid redirect loops:
+    // TODO: this won't work in safari yet:
+    //if (thisTab.httpsRedirects[e.url] >= 7) {
+    //    console.log('HTTPS: cancel https upgrade. redirect limit exceeded for url: \n' + e.url)
+    //    e.preventDefault()
+    //    e.target.url = thisTab.downgradeHttpsUpgradeRequest(e)
+    //}
+
+    const url = e.url
+    const isMainFrame = true // always main frame in this handler
```

Comment:
this is safari related, so maybe say that more explicitly?

Reply:
Probably not necessary, the whole file and directory is safari overrides